### ISSUE TITLE
If a page fails to load, `goto()` should return an error instead of success

### ISSUE THREAD

**Issue by Mr0grog** [segmentio/nightmare#494]
If a page fails to load, `goto()` should probably call back with an error instead of indicating success. This is a contrived example, but consider a time when the site you are trying to load is down:

```js
var Nightmare = require('nightmare');
var nightmare = Nightmare({ show: true })

nightmare
  .goto('http://this-is-not-a-real-domain.com')
  .then(function (result) {
    console.log(`Navigated successfully to this-is-not-a-real-domain.com!`);
  }, function (error) {
    console.error(`Error navigating: ${error}`);
  });

// Prints:
// Navigated successfully to this-is-not-a-real-domain.com!
```

**Comment by Mr0grog** [comment]
Happy to work on a PR for this if the general idea seems reasonable. Just let me know.

**Comment by rosshinkley** [comment]
@Mr0grog Ooh, that's not great.  I'd +1 for a PR.

**Comment by Mr0grog** [comment]
Fixed in #553.
